subsys/net/lib/utils/addr_utils.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <net/socket.h>
+
+const char *net_addr_str_find_port(const char *addr_str)
```

Comment:
This looks like it is doing similar things as what `net_ipaddr_parse()` is doing. Could you utilize that function here?

Reply:
> This looks like it is doing similar things as what net_ipaddr_parse() is doing.

Yes, I recalled that we should have a func like that and did my homework on that. So, first of all, net_ipaddr_parse() would be similar to net_getaddrinfo_addr_str(), not net_addr_str_find_port(). But it's not, because net_ipaddr_parse() only handles numeric addresses, whereas newly introduced net_getaddrinfo_addr_str() can handle domain names too (and implications of that, like the fact that domain names may resolve to multiple underlying addresses).

I agree that such perceived duplication is not ideal, but as explained, I did my homework on it. My idea of dealing with that is along the lines of my previous replies - treat existing functions like net_ipaddr_parse() as IP-stack level (I believe it's used (only?) to parse DNS server address, which has to be done on the stack level). Whereas idea here is to introduce a set of more generic application-level functions, implemented on top of POSIX API, and thus fully portable, e.g. to Linux too.

Another (parallel) outlook may be to treat functions like net_ipaddr_parse() as limited, but optimized functions, whereas net_getaddrinfo_addr_str() is generic, but has higher overhead. For example, I wouldn't propose to replace call to net_ipaddr_parse() in DNS code to net_getaddrinfo_addr_str() - it doesn't make sense to bring up full overhead of getaddrinfo(). (It might be possible to make net_ipaddr_parse() call newly introduced net_addr_str_find_port(), but looking at net_ipaddr_parse(), I'm not sure that'll give much benefit there).

Hope this explains in detail my outlook on the matter.

Reply:
I propose following:
* Moving generic networking related helpers to `net/utils.h` from `net/net_ip.h`. The latter is not a perfect place for utility functions anyway.
* Modifying `net_ipaddr_parse()` to support DNS resolving if the input string is not a IP address.

Reply:
> Moving generic networking related helpers to net/utils.h from net/net_ip.h.

Minor, but I named it net/netutils.h to have better namespacing. Less minor, that I'd really like to keep it POSIX API based (and buildable). One of the usecases is pulling those files (netutils.h and the implementation) to another POSIX-compatible system and using there.

> Modifying net_ipaddr_parse() to support DNS resolving if the input string is not a IP address.

But why? Again, in the current codebase, only DNS initialization uses it (well, log_backend_net.c also calls it, but that can be considered just the same system-level). And DNS initialization requires a numeric address, so supporting domain names there is useless. And you can't easily modify net_ipaddr_parse() to deal with domain names, because then it needs to be modified to handle case of multiple IP addresses per domain name (e.g. IPv4 vs IPv6 addresses, or just multiple addresses at all). And if you don't support multiple return addresses, it remains a toy function, not suitable for all possible real-world uses, but which however was bloated up beyond its real usecase.